modules/openapi-generator/src/main/resources/typescript/model/ObjectSerializer.mustache

```diff
@@ -204,14 +233,15 @@ export class ObjectSerializer {
         }
 
         const normalMediaTypes = mediaTypes.map(this.normalizeMediaType);
-        let selectedMediaType: string | undefined = undefined;
-        let selectedRank: number = -Infinity;
-        for (const mediaType of normalMediaTypes) {
-            if (supportedMediaTypes[mediaType!] > selectedRank) {
-                selectedMediaType = mediaType;
-                selectedRank = supportedMediaTypes[mediaType!];
-            }
-        }
+
+        const supportedAndOrderedMediaTypes = supportedMimeTypePredicatesWithPriority.reduce<string[]>((result, predicate) => {
```

Comment:
since only `supportedAndOrderedMediaTypes[0]` will be used, how about a for loop with early break on the first result instead of compiling the entire list to simplify the code a bit?

Reply:
@macjohnny, that's a fair point. I'll update the implementation. Thanks! 👍

Reply:
Updated